src/Microsoft.PowerShell.Commands.Utility/commands/utility/Join-String.cs

```diff
@@ -172,28 +172,28 @@ public sealed class SeparatorArgumentCompleter : IArgumentCompleter
         "`r`n";
 #endif
 
-        private static readonly Dictionary<string, (string Tooltip, string ListItemText)> s_separatorMappings = new(capacity: 7)
+        private static readonly CompletionHelpers.CompletionDisplayInfoMapper SeparatorDisplayInfoMapper = separator => separator switch
         {
-            { ",", (TabCompletionStrings.SeparatorCommaToolTip, "Comma") },
-            { ", ", (TabCompletionStrings.SeparatorCommaSpaceToolTip, "Comma-Space") },
-            { ";", (TabCompletionStrings.SeparatorSemiColonToolTip, "Semi-Colon") },
-            { "; ", (TabCompletionStrings.SeparatorSemiColonSpaceToolTip, "Semi-Colon-Space") },
-            { NewLineText, (StringUtil.Format(TabCompletionStrings.SeparatorNewlineToolTip, NewLineText), "Newline") },
-            { "-", (TabCompletionStrings.SeparatorDashToolTip, "Dash") },
-            { " ", (TabCompletionStrings.SeparatorSpaceToolTip, "Space") },
+            "," => new(TabCompletionStrings.SeparatorCommaToolTip, "Comma"),
+            ", " => new(TabCompletionStrings.SeparatorCommaSpaceToolTip, "Comma-Space"),
+            ";" => new(TabCompletionStrings.SeparatorSemiColonToolTip, "Semi-Colon"),
+            "; " => new(TabCompletionStrings.SeparatorSemiColonSpaceToolTip, "Semi-Colon-Space"),
+            "-" => new(TabCompletionStrings.SeparatorDashToolTip, "Dash"),
+            " " => new(TabCompletionStrings.SeparatorSpaceToolTip, "Space"),
+            _ when separator == NewLineText => new(StringUtil.Format(TabCompletionStrings.SeparatorNewlineToolTip, NewLineText), "Newline"),
```

Comment:
This one was awkward since `switch` requires compile time constant so `NewLineText` couldn't work like the others. 

Can always just use `if` conditions to get around this.

Reply:
Can we convert NewLineText to const? Why is it static?

Reply:
@iSazonov good catch. I missed that one, fixed in fee92b401ce1449741c3fa31dca6ca507b45fbda